xpdan/data_reduction.py

```diff
@@ -0,0 +1,382 @@
+#!/usr/bin/env python
+##############################################################################
+#
+# xpdacq            by Billinge Group
+#                   Simon J. L. Billinge sb2896@columbia.edu
+#                   (c) 2016 trustees of Columbia University in the City of
+#                        New York.
+#                   All rights reserved
+#
+# File coded by:    Timothy Liu
+#
+# See AUTHORS.txt for a list of people who contributed.
+# See LICENSE.txt for license information.
+#
+##############################################################################
+#from dataportal import DataBroker as db
+#from dataportal import get_events, get_table, get_images
+#from metadatastore.commands import find_run_starts
+
+import os
+import warnings
+import datetime
+import yaml
+import numpy as np
+import tifffile as tif
+import matplotlib as plt
+from time import strftime
+from unittest.mock import MagicMock
+
+from .glbl import an_glbl
+from .utils import _clean_info, _timestampstr
+
+from pyFAI.azimuthalIntegrator import AzimuthalIntegrator
+
+# top definition for minimal impacts on the code 
+if an_glbl._is_simulation:
+    db = MagicMock()
+    get_events = MagicMock()
+    get_images = MagicMock()
+else:
+    from databroker.databroker import get_table
+    from databroker.databroker import DataBroker as db
+    from databroker import get_images
+    from databroker import get_events
+
+w_dir = os.path.join(an_glbl.home, 'tiff_base')
+W_DIR = w_dir # in case of crashes in old codes
+
+
+
+class DataReduction:
+    """ class that handle operations on images from databroker header
+
+        Note: not a callback
+    """
+    def __init__(self, image_field=None):
+        # for file name 
+        self.fields = ['sample_name','sp_type', 'sp_requested_exposure']
+        self.labels = ['dark_frame']
+        self.data_fields = ['temperature']
+        self.root_dir_name = 'sample_name'
+        if image_field is None:
+            self.image_field = an_glbl.det_image_field
+
+    def _feature_gen(self, event):
+        ''' generate a human readable file name.
+
+        file name is generated by metadata information in event
+        run_start
```

Comment:
file name will include ``motor`` read back value which is recorded at event level